content/projects/oop-for-data-science/_index.md

```diff
@@ -1,3 +1,37 @@
 ---
 title: OOP for data science
+ready: False
 ---
+
+## Prerequisite
+
+You need to complete the Multivariate Regression assignment using the salary dataset before doing this assignment ({{% contentlink "projects/data-science-specific/multivariate-linear-regression" %}}). You will use the data and the model you built in that assignment for this OOP assignment.
+
+## Assignment
+It is efficient to put machine learning models and other data science techniques into classes so that we can reuse them later and change attributes without changing the code behind these models. Independent concepts can also be put into independent classes: for example, the functioning of a cross-validate class should not affect the functioning of a linear regression class. These concepts are also referred to as high cohesion and low coupling.
+
+1. Create a class called `Metrics` that has methods to compute the Mean Squared Error, Root Mean Squared Error, R-squared and Adjusted R-squared. Name these methods `mse`, `rmse`, `r_squared` and `adj_r_squared`. The class should have the following parameters:
```

Comment:
When writing code about any domain it's good to think about the language of that domain as spoken by humans to humans. 

If I had a class named Apples and I needed to talk to my team mates about the use of the class I would say something like this: "So let's say I make three Appleses". It would be better if I called it `Apple` so the english makes sense. 

Calling this class `Metric` also seems wrong though. What is it? `MetricCalculator` maybe?

`Metrics` would be a good name if it was not a class that was meant to be instantiated. Which happens sometimes but we usually use modules for that in Python.

Reply:
changed class names